sky/provision/gcp/instance_utils.py

```diff
@@ -650,23 +650,32 @@ def create_instances(
 
         all_names = []
         if 'reservationAffinity' in config:
+            specific_reservations = set(config['reservationAffinity']['values'])
             reservations = gcp_cloud.GCP().get_reservations_available_resources(
                 config['machineType'],
                 region=zone.rpartition('-')[0],
                 zone=zone,
-                specific_reservations=set(
-                    config['reservationAffinity']['values']))
+                specific_reservations=specific_reservations)
+            # Filter the reservations by the user-specified ones, because
+            # reservations contains auto reservations as well, which does not
+            # need to explicitly specify in the config for creating instances.
+            specific_reservations_to_count = {
+                reservation: count
+                for reservation, count in reservations.items()
+                if reservation in specific_reservations
+            }
```

Comment:
```suggestion
            # Filter the reservations by the user-specified ones, because
            # reservations contain auto reservations as well, which do not
            # need to explicitly specify in the config for creating instances.
            specific_reservations_to_count = {
                reservation: count
                for reservation, count in reservations.items()
                if reservation in specific_reservations
            }
```